ddtrace/appsec/processor.py

```diff
@@ -0,0 +1,95 @@
+import json
+import os.path
+from typing import Any
+from typing import ClassVar
+from typing import Optional
+from typing import TYPE_CHECKING
+
+import attr
+
+import ddtrace
+from ddtrace import config
+from ddtrace.appsec._ddwaf import DDWaf
+from ddtrace.constants import MANUAL_KEEP_KEY
+from ddtrace.internal.logger import get_logger
+from ddtrace.utils.formats import get_env
+
+
+if TYPE_CHECKING:
+    from ddtrace import Span
+
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+DEFAULT_RULES = os.path.join(ROOT_DIR, "rules.json")
+
+log = get_logger(__name__)
+
+
+@attr.s(eq=False)
+class AppSecProcessor(object):
+
+    tracer = attr.ib(type=ddtrace.Tracer, default=None)
+    rules = attr.ib(type=str, default=None)
+    _ddwaf = attr.ib(type=DDWaf, default=None)
+
+    enabled = False
+    _instance = None  # type: ClassVar[Optional[AppSecProcessor]]
+
+    def __attrs_post_init__(self):
+        # type: () -> None
+        if self.tracer is None:
+            self.tracer = ddtrace.tracer
+        if self.rules is None:
+            self.rules = get_env("appsec", "rules", default=DEFAULT_RULES)
+        if self._ddwaf is None:
+            with open(self.rules, "r") as f:
+                rules = json.load(f)
+            self._ddwaf = DDWaf(rules)
```

Comment:
ditto

Reply:
this one is initialized from another attribute (`self.rules`)